File: tutorial/cpp/stl/containers/stack/stack.cpp
// SPDX-License-Identifier: BSD-3-Clause

#include <iostream> // cout
#include <stack> // stack
using namespace std;

int main() {
    stack<int> st;
    for (int i = 0; i < 5; ++i) {
        st.push(i);
    }

    while (!st.empty()) {
        auto top = st.top(); // get the top element
        st.pop(); // remove the top element
        cout << top << " "; // 4 3 2 1 0
    }
    cout << "\n";

    return 0;
}

File: tutorial/rust/std/containers/stack/stack.rs
// SPDX-License-Identifier: BSD-3-Clause
// Run: rustc stack.rs && ./stack

fn main() {
    let mut st: Vec<i32> = Vec::new();
    st.push(1);
    st.push(2);
    st.push(3);

    while let Some(top) = st.pop() {
        println!("{}", top);
    }
    // print: 3 2 1
}
